src/main/java/com/Studentlifr/GratitudeBoard/security/SecurityConfiguration.java

```diff
@@ -0,0 +1,119 @@
+
+    public SecurityConfiguration(@Value("${jwt.key}") String jwtKey, UserDetailsServiceImpl userDetailsService) {
+        this.jwtKey = jwtKey;
+        this.userDetailsService = userDetailsService;
+        }
+        @Autowired
+        @Lazy
+        public void setAccessDeniedHandler(AccessDeniedHandler accessDeniedHandler) {
+        }
+    @Bean
+    public AccessDeniedHandler accessDeniedHandler() {
+        return new AccessDeniedHandlerImpl();
+    }
+    @Bean
+    CorsConfigurationSource corsConfigurationSource() {
+        UrlBasedCorsConfigurationSource source = new UrlBasedCorsConfigurationSource();
+        CorsConfiguration corsConfiguration = new CorsConfiguration().applyPermitDefaultValues();
+        corsConfiguration.addAllowedMethod("*");
+        corsConfiguration.addAllowedHeader("*");
+        corsConfiguration.addAllowedOrigin("*");
+        source.registerCorsConfiguration("/**", corsConfiguration);
+        log.info("CORS config {}", source.getCorsConfigurations());
+        return source;
+    }
+
+    @Bean
+    public SecurityFilterChain apiFilterChain(HttpSecurity http) throws Exception {
+        http
+                .csrf( csrf -> csrf.disable())
+                .authorizeHttpRequests(authorize -> authorize
+                        .requestMatchers(PUBLIC_URLS)
+                        .permitAll()
+                        .requestMatchers(CommonConstants.USER_URLS)
+                        .hasAnyAuthority(CommonConstants.USER_ROLE , CommonConstants.ADMIN_ROLE)
+                        .requestMatchers(CommonConstants.ADMIN_URLS)
+                        .hasAuthority(CommonConstants.ADMIN_ROLE)
+                        .anyRequest()
+                        .authenticated()
+                )
+//                .exceptionHandling(exceptionHandlingConfigurer-> exceptionHandlingConfigurer.accessDeniedHandler(accessDeniedHandler))
+                .sessionManagement(sessionManagement -> sessionManagement.sessionCreationPolicy(SessionCreationPolicy.STATELESS))
+                .oauth2ResourceServer(oauth2 -> oauth2.jwt(Customizer.withDefaults()));
+
+
+        return http.build();
+    }
+
+    @Bean
+    public BCryptPasswordEncoder bCryptPasswordEncoder() {
+        return new BCryptPasswordEncoder(9);
+    }
+    @Bean
+    JwtEncoder jwtEncoder() {
+
+        return new NimbusJwtEncoder(new ImmutableSecret<>(jwtKey.getBytes()));
+    }
+
+
+    @Bean
+    public JwtDecoder jwtDecoder() {
+        byte[] bytes = jwtKey.getBytes();
+        SecretKeySpec originalKey = new SecretKeySpec(bytes, 0, bytes.length, "RSA");
+        return NimbusJwtDecoder.withSecretKey(originalKey).macAlgorithm(MacAlgorithm.HS512).build();
+    }
+    @Bean
+    public AuthenticationManager authenticationManagerBean() {
+        var authProvider = new DaoAuthenticationProvider();
+        authProvider.setUserDetailsService(userDetailsService);
+        authProvider.setPasswordEncoder(bCryptPasswordEncoder());
```

Comment:
Please override the default settings in `bCryptPasswordEncoder` [docs](https://docs.spring.io/spring-security/site/docs/current/api/org/springframework/security/crypto/bcrypt/BCryptPasswordEncoder.html#%3Cinit%3E(int,java.security.SecureRandom))


and read the configuration settings from `application.properties`

Reply:
Could not able to understand what exactly required!